sigstore/_sign.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import hashlib
+
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+
+from sigstore._internal.fulcio import (
+    FulcioCertificateSigningRequest,
+    FulcioClient,
+)
+from sigstore._internal.oidc import Identity
+from sigstore._internal.rekor import RekorClient
+
+
+def _no_output(*a, **kw):
+    pass
+
+
+def sign(file_, identity_token, output=_no_output):
```

Comment:
I think we should figure out what the return type should be soon. I'm thinking something like a `SigningResult` which allows you to tell between success/failure and whether there is a reason.

We could also just not have a return and consider any non-throwing call to `sign` to be a success.

I'll go ahead and merge this anyway since we have too many pieces of work branched off this, but we should definitely revisit it.